private/bufpkg/bufcheck/bufbreaking/internal/bufbreakingcheck/bufbreakingcheck.go

```diff
@@ -783,9 +805,14 @@ func checkOneofNoDelete(add addFunc, corpus *corpus, previousMessage protosource
 	if err != nil {
 		return err
 	}
-	for previousName := range previousNameToOneof {
+	for previousName, previousOneof := range previousNameToOneof {
 		if _, ok := nameToOneof[previousName]; !ok {
-			add(message, nil, message.Location(), `Previously present oneof %q on message %q was deleted.`, previousName, message.Name())
+			// If this is a synthetic oneof, deleting of the oneof does not signal that a oneof was deleted,
+			// instead it signals that the field was renamed or changed to optional, which are done in
```

Comment:
Did you mean changed to non-optional?

Reply:
This `if` statement says that if previously we were non-optional and now are optional, so this comment is right, changed to optional

Reply:
```
If this is a synthetic oneof [...] the field was [...] changed to optional.
```
A synthetic oneof _is_ an optional. 

Additionally, if we were previously non-optional, then we'll fall into the `if` and will report an annotation, whereas I think your comment was meant to capture the cases that should _not_ fall into the `if`.